### PR TITLE
Add GitHub Actions workflow for Jekyll site build and deploy

### LINKED ISSUE #507: Pull latest `gorefs.yaml` file before building and deploying to GitHub Pages
To be done after https://github.com/geneontology/geneontology.github.io/issues/506.

- [ ] In order to customize our deployment process we'll need to switch from a [branch-based](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-from-a-branch) deployment to an [action-based](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-from-a-branch) one. Jeyll provides a [starter workflow](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml) as a baseline. Start by adding a copy of that starter to this repo in `.github/workflows/deploy.yaml`
- [ ] Add a step to the `build` which runs the mechanism (from #506) to pull the latest `gorefs.yaml` into `_data` before running `jekyll build`.
- [ ] Change the [settings](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow) in this repo to have GitHub pages use GitHub Actions instead of deploying from a branch.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
Fixes #507

### Summary

* Add a GitHub Actions workflow (`.github/workflows/deploy.yaml`) which fetches the latest GO REF data (`make _data/gorefs.yaml`) and then builds and deploys the Jekyll site. 
* Add the `vendor` directory which is created by the `ruby/setup-ruby` action (https://github.com/ruby/setup-ruby?tab=readme-ov-file#bundle-config) to the `exclude` list in the Jekyll config.
* Remove a URL hash from the `redirect_from` in `_docs/taxon-constraints.md`. I don't really understand how this was working before in the automatic Jekyll build provided by GitHub Pages. This hash caused the build to fail for me locally as well as when I tested this deployment in a fork: https://github.com/pkalita-lbl/geneontology.github.io/actions/runs/8302393139/job/22724573998#step:6:373

### Comments

Just before merging this we will need to update this repo's settings to change the GitHub Pages build and deploy source to GitHub Actions. I don't have permission to make that change in this repo. Maybe @kltm can do it. Here's a screenshot of the correct setting in my test fork:

image

### CODE REVIEW

**Comment by kltm** [comment]
@pkalita-lbl You should now have permissions to make the changes you need.

File: _docs/taxon-constraints.md
Comment (kltm) [inline _docs/taxon-constraints.md:5]: Hm, this does not seem to work?

**Review by kltm — COMMENTED** [review]
This looks pretty much fine to me.

Does there need to be an update in end-user (i.e. for curators and non-technical "site maintainers") docs?

File: _docs/taxon-constraints.md
Comment (pkalita-lbl) [reply _docs/taxon-constraints.md:5]: Yeah, I also noticed that the redirect doesn't seem to work on the production site currently -- with or without the hash. Other pages with `redirect_from` items seem to work okay. I'm not sure what's special about this one. Nonetheless it's definitely orthogonal to these changes. So maybe a new issue to investigate is warranted?

**Comment by pkalita-lbl** [comment]
> Does there need to be an update in end-user (i.e. for curators and non-technical "site maintainers") docs?

I don't think so. From a user/editor/maintainer perspective the site will build and redeploy on pushes to `master` just like it used to. 

> You should now have permissions to make the changes you need.

Yes I do! I will take care of the settings before merging.

File: _docs/taxon-constraints.md
Comment (kltm) [reply _docs/taxon-constraints.md:5]: Fair enough--I couldn't find a way to make it work. As we've had no complaint (that I know of), it's more an oddity than anything else.

**Review by kltm — APPROVED** [review]
I think this looks good to me.

**Pull request merged**
